pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -149,17 +154,17 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 
 	err = dfutil.ValidateK8sResourceName("component name", a.ComponentName)
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "unable to validate component name")
 	}
 
 	err = dfutil.ValidateK8sResourceName("component namespace", parameters.EnvSpecificInfo.GetNamespace())
 	if err != nil {
-		return err
+		return errors.Wrapf(err, "unable to validate component namespace")
 	}
 
 	pushDevfileCommands, err := common.ValidateAndGetPushDevfileCommands(a.Devfile.Data, a.devfileBuildCmd, a.devfileRunCmd)
 	if err != nil {
-		return fmt.Errorf("failed to validate devfile build and run commands: %w", err)
+		return errors.Wrapf(err, "failed to validate devfile build and run commands")
```

Comment:
We have decided to use the native `fmt.Errorf()` with `%w` instead of `errors.Wrapf` (more info in https://go.dev/blog/go1.13-errors)